src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -813,13 +831,69 @@ constexpr static const auto EXPORT_IMPORT = "EXPORT_IMPORT";  //!< Device suppor
 
 }  // namespace device
 
+namespace property {
+struct PropertyNames {
+    /**
+     * @brief Constructs property
+     * @param property property
+     * @return Pair of string key representation and type erased property value.
+     */
+    template <typename T, PropertyMutability mutability>
+    inline std::pair<std::string, Any> operator()(const ov::Property<T, mutability>& property) const {
+        return {name(), std::vector<ov::PropertyName>{property.name()}};
+    }
+    /**
+     * @brief Constructs property
+     * @param property_name property name
+     * @return Pair of string key representation and type erased property value.
+     */
+    inline std::pair<std::string, Any> operator()(const ov::PropertyName& property_name) const {
+        return {name(), std::vector<ov::PropertyName>{property_name}};
+    }
+    /**
+     * @brief Constructs property
+     * @param property_names property names
+     * @return Pair of string key representation and type erased property value.
+     */
+    inline std::pair<std::string, Any> operator()(const std::vector<ov::PropertyName>& property_names) const {
+        return {name(), property_names};
+    }
+    static std::string name() {
+        return std::string("PROPERTY_NAMES");
+    }
+};
+/**
+ * @brief Read-only property to specify inserested list of properties
+ * @ingroup ie_dev_api_plugin_api
+ */
+static constexpr PropertyNames names;
+}  // namespace property
+
+namespace subgraph {
+struct SubgraphId : public Property<uint32_t, PropertyMutability::RO> {
+public:
+    using Property<uint32_t, PropertyMutability::RO>::Property;
+    /**
+     * @brief Constructs property
+     * @param id subgraph id
+     * @return Pair of string key representation and type erased property value.
+     */
+    inline std::pair<std::string, Any> operator()(const uint32_t& id) const {
+        return {std::string(name()), id};
+    }
+};
+static constexpr SubgraphId id{"SUBGRAPH_ID"};
+static constexpr Property<std::map<uint32_t, ov::AnyMap>, PropertyMutability::RO> properties{"SUBGRAPH_PROPERTIES"};
```

Comment:
what is the purpose for end users of having information from subgraph ?

BTW, maybe it's better to use unordered_map ?